metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -222,24 +222,47 @@ def _next_unique_table_alias(self) -> str:
         """Return the next unique table alias to use in generating queries."""
         return SequentialIdGenerator.create_next_id(StaticIdPrefix.SUB_QUERY).str_value
 
+    def _choose_time_spine_source(
+        self, agg_time_dimension_instances: Tuple[TimeDimensionInstance, ...]
+    ) -> TimeSpineSource:
+        """Determine which time spine source to use when building time spine dataset.
+
+        Will choose the time spine with the largest granularity that can be used to get the smallest granularity requested
+        in the agg time dimension instances. Example:
+        - Time spines available: SECOND, MINUTE, DAY
+        - Agg time dimension granularity needed for request: HOUR, DAY
+        --> Selected time spine: MINUTE
+        """
+        assert (
+            agg_time_dimension_instances
+        ), "Building time spine dataset requires agg_time_dimension_instances, but none were found."
+        smallest_agg_time_grain = min(dim.spec.time_granularity for dim in agg_time_dimension_instances)
```

Comment:
Ok, this works but it'll start failing with custom granularities. Good thing for me to note here - we'll need to be able to go from the agg time dimension instance to the time spine index granularity and check up on that.